pkg/input/provider/dedupe/dedupe.go

```diff
@@ -0,0 +1,102 @@
+// Package dedupe implements a duplicate URL deduplication mechanism
```

Comment:
We can consider using https://github.com/projectdiscovery/utils/blob/main/dedupe/dedupe_test.go